### PR TITLE
Create `CsrfToken` model and utilities

### LINKED ISSUE #12271: Create CSRF Token model and utilities

### PR DESCRIPTION
**Pull request by bunchesofdonald**
Creates a `CsrfToken` model and associated utilities to store CSRF tokens associated with a client identifier. 

Closes #12271 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [x] This pull request includes helpful docstrings.
- [x] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Review by znicholasbrown — APPROVED** [review]
I'm not familiar with how the docs for this should work but the rest of it looks great!

**Review by serinamarie — APPROVED** [review]
🔏

File: src/prefect/server/models/csrf_token.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    if result.rowcount < 1:
+        # Client does not have a token, create a new one
+        model = db.CsrfToken(
+            client=client,
+            token=token,
+            expiration=expiration,
+        )
+        session.add(model)
+        await session.flush()
```
[the remark this answers is not in the document]
Comment (bunchesofdonald) [reply src/prefect/server/models/csrf_token.py:52]: I was just going for maximum sqlite compatibility, I didn't realize we were using on_conflict over here in OSS, I can update.

**Review by serinamarie — APPROVED** [review]
API ref in docs looks good
Screenshot 2024-03-14 at 10 04 17 AM

**Pull request merged**